pkg/convert/list.go

```diff
@@ -70,20 +70,15 @@ func (enc *listEncoder) fromPropertyValue(p resource.PropertyValue) (tftypes.Val
 }
 
 func (dec *listDecoder) toPropertyValue(v tftypes.Value) (resource.PropertyValue, error) {
-	if !v.IsKnown() {
-		return unknownProperty(), nil
-	}
-	if v.IsNull() {
-		return resource.NewPropertyValue(nil), nil
-	}
```

Comment:
This sequence needs to head every call to `toPropertyValue` for a correct implementation. Instead of adding it to the couple places it was missing, I moved the block to `decode`.